### ISSUE TITLE
`bun install` fails

### ISSUE THREAD

**Issue by svercl** [oven-sh/bun#88]
I ran bun install with some packages installed and when I run `bun install` it fails.
Packages in my `package.json` (other than the defaults with `bun create next ..`
- @mantine/core 3.4.3
- @mantine/hooks 3.4.3
- @mantine/next 3.4.3

One interesting thing I noticed is the package names seem to have been corrupted along the way.

It fails with this error:
```
bun install v0.0.59
  🔍 Resolving... GET https://registry.npmjs.org/ :rea/-/ :rea-0.8.0.tgz - 400
  🔍 Resolving... GET https://registry.npmjs.org/V/-/V-3.3.2.tgz - 400
  🔍 Resolving... GET https://registry.npmjs.org/ct-is16.3.0-alph/-/ct-is16.3.0-alph-0.7.5.tgz - 404
  🔍 Resolving... GET https://registry.npmjs.org/!!0/-/!!0-1.1.0.tgz - 400
  🔍 Resolving... GET https://registry.npmjs.org/@radix-ui/reacin/env bun
/-/reacin/env bun
-0.1.0.tgz - 400
  🔍 Resolving... GET https://registry.npmjs.org/   areact/-/   areact-1.0.0.tgz - 400
  🔍 Resolving... GET https://registry.npmjs.org/  :"  :/-/  :"  :-1.0.2.tgz - 400
  🔍 Resolving... Failed to download package tarball for package StringOrTinyString{ .remainder_buf = { 49, 10, 98, 117, 110, 98, 117, 110, 98, 99, 58, 0, 0, 29, 0, 0, 0, 81, 2, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0 }, .remainder_len = 14, .is_tiny_string = 1 }
  🔍 Resolving... GET https://registry.npmjs.org/!0Q/-/!0Q-0.2.5.tgz - 400
  🔍 Resolving... GET https://registry.npmjs.org/a.1^16.0.0 || 16./-/a.1^16.0.0 || 16.-0.7.5.tgz - 404
  🔍 Resolving... GET https://registry.npmjs.org/-is ]a   #!/-/-is ]a   #!-0.1.2.tgz - 400
  🔍 Resolving... GET https://registry.npmjs.org/st-cache-v0.0./-/st-cache-v0.0.-11.7.1.tgz - 404

 4 packages installed [476.00ms]
```

**Comment by Jarred-Sumner** [comment]
I'm not sure yet how to reproduce this. It's some kind of undefined memory bug. If you do `rm -rf ~/.bun/install` and re-install, does it happen again? If so, can you send your package.json?

**Comment by svercl** [comment]
I modified my `package.json` manually added the packages with `@`. It's semi-related to #89. I'll see if I can reproduce, but I'm pretty sure it's resolved.

**Comment by svercl** [comment]
I retraced my steps on `0.0.63` and I'm faced with a segfault now. It looked fine at first, showed progress and the usual install process, but then I'm met with that. Additionally, it did not install the package. However, Re-running `bun install` again (the second time) worked!

**Comment by svercl** [comment]
Additionally, I just tried with another fresh project and it appeared to work, but it missed some packages this time around.

**Comment by Jarred-Sumner** [comment]
I believe the cause is incorrect state machine logic in the HTTP client
that causes undefined memory  when the network is saturated. I am most able
to reproduce it when I send a batch of about 10,000 http requests in quick
succession on localhost. Working on a fix

**Comment by Jarred-Sumner** [comment]
This might have been fixed or it might not have been fixed. I definitely fixed some cases where this happens, but I am not confident all cases are fixed. Please let me know if you see it again

**Comment by svercl** [comment]
Will do!

**Comment by svercl** [comment]
I walked through my steps again and it's the problem; in fact it is almost identical output

**Comment by Jarred-Sumner** [comment]
@bradms can you give this another try with Bun v0.0.69?

**Comment by svercl** [comment]
Hmm it looks like the corruption is resolved, but it is impossible to install @ packages for me on bun 0.0.70. (The same was true for bun 0.0.69 as well)

**Comment by svercl** [comment]
Wait, this seems to have worked now. I ran add with the @ packages and they ended up working this time, but did not before.

Unfortunately, it is not starting the web server. It does say it is running but it does not appear to be.

**Comment by Jarred-Sumner** [comment]
> Unfortunately, it is not starting the web server. It does say it is running but it does not appear to be.

Can you give a little more info? If the default port (3000) is already in use, it tries `3001` - `3009` until it finds a match and it reports which one it actually is running at in the log

**Comment by Jarred-Sumner** [comment]
> Hmm it looks like the corruption is resolved, but it is impossible to install @ packages for me on bun 0.0.70. (The same was true for bun 0.0.69 as well)
> Wait, this seems to have worked now. I ran add with the @ packages and they ended up working this time, but did not before.

I believe this is fixed by 5d51b4c3972c7d0733a54ba2038721ce3236a92a and fce94e29a00f7b0b533c2725fe36b807a61ea517. However, the lockfile format will be invalidated in the next release

**Comment by svercl** [comment]
> If the default port (3000) is already in use, it tries 3001 - 3009


Yeah, that is kind of what I figured, but I remember clicking the link it spit out and that lead me to a blank page which was trying to load but never did. I think my system was having issues at that time with network. (I do not blame `bun` here, because it was probably on my end)

> next release


I look forward to it!

**Comment by Jarred-Sumner** [comment]
This was fixed in v0.0.71. Please re-open if that is not the case
